### ISSUE TITLE
Check failed: old_node_input_size >= old_node_input_slots (4 vs. 3) 

### ISSUE THREAD

**Issue by weijianghn** [tensorflow/tensorflow#74303]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

master

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Bazel version

6.5.0

### Current behavior?

run keras example code
Simple MNIST convnet
batch_size = 128
epochs = 15

model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])

model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, validation_split=0.1)
coredump and shows:
F tensorflow/core/common_runtime/mkl_layout_pass.cc:2490] Check failed: old_node_input_size >= old_node_input_slots (4 vs. 3)

### Standalone code to reproduce the issue

```shell
import numpy as np
import keras
from keras import layers
# Model / data parameters
num_classes = 10
input_shape = (28, 28, 1)

# Load the data and split it between train and test sets
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

# Scale images to the [0, 1] range
x_train = x_train.astype("float32") / 255
x_test = x_test.astype("float32") / 255
# Make sure images have shape (28, 28, 1)
x_train = np.expand_dims(x_train, -1)
x_test = np.expand_dims(x_test, -1)
print("x_train shape:", x_train.shape)
print(x_train.shape[0], "train samples")
print(x_test.shape[0], "test samples")


# convert class vectors to binary class matrices
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)
model = keras.Sequential(
    [
        keras.Input(shape=input_shape),
        layers.Conv2D(32, kernel_size=(3, 3), activation="relu"),
        layers.MaxPooling2D(pool_size=(2, 2)),
        layers.Conv2D(64, kernel_size=(3, 3), activation="relu"),
        layers.MaxPooling2D(pool_size=(2, 2)),
        layers.Flatten(),
        layers.Dropout(0.5),
        layers.Dense(num_classes, activation="softmax"),
    ]
)

model.summary()
batch_size = 128
epochs = 15

model.compile(loss="categorical_crossentropy", optimizer="adam", metrics=["accuracy"])

model.fit(x_train, y_train, batch_size=batch_size, epochs=epochs, validation_split=0.1)
```

**Comment by weijianghn** [comment]
I follow the build instructions from the webpage, and use bazel build the src without intel mkl option.  
bazel build //tensorflow/tools/pip_package:wheel --repo_env=WHEEL_NAME=tensorflow_cpu --config=dbg
After I install the package,  I find that MKL is enabled.  that's the failure reason.

**Comment by Venkat6871** [comment]
Hi **@weijianghn** ,
Apologies for the delay. I tried to run your code on Colab using TF v2.17.0 and nightly. It is working fine for me. Please find the [gist](https://colab.research.google.com/gist/Venkat6871/97a9be9be4121a30716a7703cb874009/74303_2-17-nightly-v.ipynb) here for reference. If i am wrong please let me know.
Thank you!

**Comment by parveenduhan** [comment]
I am also facing the same issue. I have to disable TF_ENABLE_ONEDNN to get rid of this which I don't want.
